src/crowbar.mli

```diff
@@ -1,52 +1,128 @@
+    a smaller domain, consider using {!range}. *)
+
+val uint8 : int gen
+(** [uint8] generates an unsigned byte, ranging from 0 to 255 inclusive. *)
+
+val int8 : int gen
+(** [int8] generates a signed byte, ranging from -128 to 127 inclusive. *)
+
+val int32 : Int32.t gen
+(** [int32] generates a 32-bit signed integer. *)
+
+val int64 : Int64.t gen
+(** [int64] generates a 64-bit signed integer. *)
+
+val float : float gen
+(** [float] generates a double-precision floating-point number. *)
+
+val bytes : string gen
+(** [bytes] generates a string of arbitrary length (including zero-length strings).  *)
+
+val bool : bool gen
+(** [bool] generates a yes or no answer. *)
+
+val range : int -> int gen
+(** [range n] is a generator for integers between 0 (inclusive) and [n] (exclusive). *)
+
+(** {2:generator_functions Functions on Generators } *)
 
 val map : ('f, 'a) gens -> 'f -> 'a gen
+(** [map gens map_fn] provides a means for creating generators using other
+    generators' output.  For example, one might generate a Char.t from a
+    {!uint8}:
+    {[
+      open Crowbar
+      let char_gen : Char.t gen = map [uint8] Char.chr
+    ]}
+*)
 
 val join : 'a gen gen -> 'a gen
 
 val bind : 'a gen -> ('a -> 'b gen) -> 'b gen
 
 val unlazy : 'a gen Lazy.t -> 'a gen
+(** [unlazy gen] forces the generator [gen].  It is useful when defining
+    generators for recursive data types:
+
+    {[
+      open Crowbar
+      type a = A of int | Self of a
+      let rec a_gen = lazy (
+        choose [
+          map [int] (fun i -> A i);
+          map [(unlazy a_gen)] (fun s -> Self s);
+        ])
+      let lazy a_gen = a_gen
+    ]}
+*)
+
 val fix : ('a gen -> 'a gen) -> 'a gen
+(** [fix fn] applies the function [fn].  It is useful when defining generators
+    for recursive data types:
+
+    {[
+      open Crowbar
+      type a = A of int | Self of a
+      let rec a_gen = fix (fun a_gen ->
+          choose [
+          map [int] (fun i -> A i);
+          map [a_gen] (fun s -> Self s);
+        ])
+    ]}
+    *)
 
 val const : 'a -> 'a gen
+(** [const a] always generates [a]. *)
+
 val choose : 'a gen list -> 'a gen
+(** [choose gens] chooses a generator arbitrarily from [gens]. *)
 
 val option : 'a gen -> 'a option gen
+(** [option a] generates either [Some a] or [None]. *)
+
 val list : 'a gen -> 'a list gen
+(** [list gen] makes a generator for lists using [gen].  Lists may be empty; for
+    non-empty lists, use {!list1}. *)
+
 val list1 : 'a gen -> 'a list gen
+(** [list1 gen] makes non-empty list generators. For potentially empty lists,
+    use {!list}.*)
 
 val with_printer : 'a printer -> 'a gen -> 'a gen
+(** [with_printer printer gen] sets the printer for [gen] to [printer].  If
+    [gen] is used to create a failing test case and the test was reached by
+    calling [check_eq] without [pp] set, [printer] will be used to print the
+    failing test case. *)
```

Comment:
This sounds like `with_printer printer gen` mutates `gen`. Maybe "[with_printer printer gen] generates the same values as [gen], but they are printed using [printer]" ?